test/ecto/query/builder/select_test.exs

```diff
@@ -86,6 +87,46 @@ defmodule Ecto.Query.Builder.SelectTest do
         escape(quote do [:foo, ^:bar] end, [], __ENV__)
       end
     end
+
+    test "supports aliasing a selected value with selected_as/2" do
```

Comment:
Sorry for a question out of nowhere, but what happens when `selected_as/2` is used two times in the same query and for the same name?

Reply:
Oh it will raise: https://github.com/elixir-ecto/ecto/pull/3966/files#diff-a4782321bba4a264f78ae4d81bc9be152d20af799129133dd078d59f2c1fb4d0R464.

Actually thank you for asking. I wanted to add a test for this but the error ^ happens before the test starts. I meant to ask for advice testing it.